common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -109,7 +109,7 @@ public Collation(
   private static final Collation[] collationTable = new Collation[4];
   private static final HashMap<String, Integer> collationNameToIdMap = new HashMap<>();
 
-  public static final int DEFAULT_COLLATION_ID = 0;
+  public static final int SPARK_INTERNAL_COLLATION_ID = 0;
```

Comment:
Can you call this simply `UTF8_BINARY_COLLATION_ID`? I think adding any other adjective will only bring confusion (what does `INTERNAL` mean?).

Also, please rename `LOWERCASE_COLLATION_ID`. It should be `UTF_BINARY_LCASE_COLLATION_ID`.